File: app/src/main/java/com/sevtinge/cemiuiler/module/home/folder/FolderColumns.java
package com.sevtinge.cemiuiler.module.home.folder;

import android.view.ViewGroup;
import android.widget.GridView;

import com.sevtinge.cemiuiler.module.base.BaseHook;
import com.sevtinge.cemiuiler.utils.DisplayUtils;
import com.sevtinge.cemiuiler.utils.Helpers;

import de.robv.android.xposed.XposedHelpers;

public class FolderColumns extends BaseHook {

    @Override
    public void init() {
        findAndHookMethod("com.miui.home.launcher.Folder", "onFinishInflate", new Helpers.MethodHook() {
            @Override
            protected void after(final MethodHookParam param) {

                GridView mContent = (GridView) XposedHelpers.getObjectField(param.thisObject, "mContent");
                int columns = mPrefsMap.getInt("home_folder_columns", 3);
                int mVerticalPadding = mPrefsMap.getInt("home_folder_vertical_padding", 0);
                mContent.setNumColumns(columns);
                if (mVerticalPadding > 0) mContent.setVerticalSpacing(DisplayUtils.dip2px(mContent.getContext(), mVerticalPadding));
                if (columns > 3 && mPrefsMap.getBoolean("home_folder_space")) {
                    ViewGroup mBackgroundView = (ViewGroup)XposedHelpers.getObjectField(param.thisObject, "mBackgroundView");
                    if (mBackgroundView != null) {
                        mBackgroundView.setPadding(
                                mBackgroundView.getPaddingLeft() / 3,
                                mBackgroundView.getPaddingTop(),
                                mBackgroundView.getPaddingRight() / 3,
                                mBackgroundView.getPaddingBottom()
                        );
                    }
                }
            }
        });

        hookAllMethods("com.miui.home.launcher.Folder", "bind", new Helpers.MethodHook() {
            @Override
            protected void after(final MethodHookParam param) {
                if (!mPrefsMap.getBoolean("home_folder_width")) return;
                GridView mContent = (GridView)XposedHelpers.getObjectField(param.thisObject, "mContent");
                ViewGroup.LayoutParams layoutParams = mContent.getLayoutParams();
                layoutParams.width = ViewGroup.LayoutParams.MATCH_PARENT;
                mContent.setPadding(0,0,0,0);
                mContent.setLayoutParams(layoutParams);
            }
        });
    }
}

File: app/src/main/java/com/sevtinge/cemiuiler/module/home/folder/FolderColumns.kt
package com.sevtinge.cemiuiler.module.home.folder

import android.view.ViewGroup
import android.widget.GridView
import com.github.kyuubiran.ezxhelper.utils.loadClass
import com.sevtinge.cemiuiler.module.base.BaseHook
import com.sevtinge.cemiuiler.utils.hookAfterAllMethods
import de.robv.android.xposed.XposedHelpers

object FolderColumns : BaseHook() {
    override fun init() {

        val value = mPrefsMap.getInt("home_folder_columns", 3)
        if (value == 3) return
        loadClass("com.miui.home.launcher.Folder").hookAfterAllMethods(
            "bind"
        ) {
            val columns: Int = value
            val mContent = XposedHelpers.getObjectField(it.thisObject, "mContent") as GridView
            mContent.numColumns = columns
            if (mPrefsMap.getBoolean("home_folder_width") && (columns > 3)) {
                mContent.setPadding(0, 0, 0, 0)
                val lp = mContent.layoutParams
                lp.width = ViewGroup.LayoutParams.MATCH_PARENT
                mContent.layoutParams = lp
            }
            if (columns > 3) {
                val mBackgroundView = XposedHelpers.getObjectField(it.thisObject, "mBackgroundView") as ViewGroup
                mBackgroundView.setPadding(
                    mBackgroundView.paddingLeft / 3, mBackgroundView.paddingTop, mBackgroundView.paddingRight / 3, mBackgroundView.paddingBottom
                )
            }
        }

    }
}